src/desktop/components/search_bar/view.coffee

```diff
@@ -64,11 +73,38 @@ module.exports = class SearchBarView extends Backbone.View
 
   checkSubmission: (e) ->
     @hideSuggestions()
+
     return if !(e.which is 13) or @selected?
 
     unless @isEmpty()
       @trigger 'search:entered', encodeURIComponent(@$input.val())
 
+  maybeHideSpotlight: (e) ->
+    @hideSpotlight() unless e.which is 37 or e.which is 39
+
+  hideSpotlight: ->
+    @$spotlight.css('z-index', '-1')
+
+  displaySpotlightSearch: (results) ->
+    return unless @enableSpotlightAutocomplete
+
+    currentQuery = @$input.val()
+    @$spotlightInitialText.text('')
+    @$spotlightRemainingText.text('')
+    return @hideSpotlight() if currentQuery.length <= 4
+
+    return unless results.length > 0
+    firstSearchResult = new SearchResult(results[0]).get('display')
+
+    return @hideSpotlight() unless firstSearchResult.toLowerCase().indexOf(currentQuery.toLowerCase()) is 0
+
+    return @hideSpotlight() unless (currentQuery.length / firstSearchResult.length) >= 0.3
+
+    remainingText = firstSearchResult.replace(new RegExp(currentQuery, 'i'), '')
+    @$spotlightInitialText.text(currentQuery)
+    @$spotlightRemainingText.text(remainingText)
+    @$spotlight.css('z-index', '2')
+
```

Comment:
If we don't plan on tweaking this further could you add a few results tests for this method since its kinda logic heavy?

Reply:
sure, ill refactor/add some tests.